### PR TITLE
Mypy

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Add mypy as part of linting check.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Enable mypy with strict option which is usually a good default, as well as ignoring missing imports.

## Tracking Issue
https://github.com/lyft/flyte/issues/650

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/types/schema.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def as_readonly(self) -> FlyteSchema:
 
 
 class FlyteSchemaTransformer(TypeTransformer[FlyteSchema]):
-    _SUPPORTED_TYPES: typing.Dict[type : SchemaType.SchemaColumn.SchemaColumnType] = {
+    _SUPPORTED_TYPES: typing.Dict[Any, SchemaType.SchemaColumn.SchemaColumnType] = {
```
Comment (kumare3) [inline flytekit/types/schema.py:302]: should this be `Type`

**Review by wild-endeavor — APPROVED** [review]

File: setup.cfg [hunk 2]
```diff
@@ -15,6 +15,13 @@ per-file-ignores =
     *:F821
     */__init__.py: F401
 
+[mypy]
+strict_optional = True
+ignore_missing_imports = True
+files = flytekit/annotated/**/*.py,tests/flytekit/unit/annotated/**/*.py
```
Comment (kumare3) [inline setup.cfg:21]: can we also add `flytekit/taskplugins/**` and `flytekit/types/**`

**Comment by kumare3** [comment]
Thank you for the quick turnaround @honnix, 2 small nits

File: flytekit/types/schema.py (hunk 1 above)
Comment (honnix) [reply flytekit/types/schema.py:302]: I think here `Any` seems to be correct. `Type` is used to annotate class objects, but here I guess it should just be instance of some type, right?

File: setup.cfg (hunk 2 above)
Comment (honnix) [reply setup.cfg:21]: Sure

File: flytekit/types/schema.py (hunk 1 above)
Comment (honnix) [reply flytekit/types/schema.py:302]: Ah wait, you are right. I miss understood the code.

**Comment by honnix** [comment]
PTAL again and feel free to merge. I'm signing off for today.

**Pull request merged**